### PR TITLE
Allow to set the participations for a PartialEvent

### PR DESCRIPTION
**Pull request by krichprollsch**
The ability to set the participations could be need for example if we want to remove all the participations and force the export to show it.

### CODE REVIEW

File: src/Event/PartialEvent.php [hunk 1]
```diff
@@ -94,6 +96,13 @@ public function addParticipation(BaseEventParticipation $participation)
         parent::addParticipation($participation);
     }
 
+    public function setParticipations(ArrayCollection $participations)
+    {
```
Comment (Taluu) [inline src/Event/PartialEvent.php:101]: control what is in the collection, as you could have a collection of `string` for example.

Comment (Taluu) [inline src/Event/PartialEvent.php:101]: And also, you should use the `Collection` type instead, instead of `ArrayCollection`

**Review by Taluu — CHANGES_REQUESTED** [review]
Not a fan to reset the whole collection, but meh I see the use cases

**Comment by krichprollsch** [comment]
@Taluu ty for your review, I just update the PR.

File: src/Event/PartialEvent.php [hunk 2]
```diff
@@ -94,6 +97,18 @@ public function addParticipation(BaseEventParticipation $participation)
         parent::addParticipation($participation);
     }
 
+    public function setParticipations(Collection $participations)
+    {
+        foreach ($participations as $participation) {
+            if (!$participation instanceof BaseEventParticipation) {
+                throw new InvalidArgumentException(sprintf('All participations must be instance of %s', BaseEventParticipation::class));
```
Comment (Taluu) [inline src/Event/PartialEvent.php]: Oops, not php 5.4. But maybe we should bump the requirement...

**Review by Taluu — CHANGES_REQUESTED** [review]

**Comment by krichprollsch** [comment]
changed @Taluu

**Review by Taluu — APPROVED** [review]

**Comment by Taluu** [comment]
LGTM 👍

**Pull request merged**